lib/packwerk/configuration.rb

```diff
@@ -47,13 +46,33 @@ def initialize(configs = {}, config_path: nil)
 
         warn(warning)
       end
+
+      inflection_file = File.expand_path(configs["inflections_file"] || "config/inflections.yml", @root_path)
+      if configs["inflections_file"]
+        warning = <<~WARNING
+          DEPRECATION WARNING: The 'inflections_file' key in `packwerk.yml` is deprecated.
+          This value is no longer cached, and you can remove the key from `packwerk.yml`.
+          You can also delete #{configs["inflections_file"]}.
```

Comment:
Just deleting it will actually break your app 😱 

I think we want an actual upgrade guide in a `UPGRADING.md` to get from 1.x to 2.0. We can do that in a separate PR, I'm happy to draft something

Reply:
I think it's a straightforward migration (as long as we explain it well) so it's fine to just make it a breaking change

Reply:
That makes sense -- should we keep one or both warnings actually? I have the warning about the `load_paths` key to push people to have simpler config, but technically it can still exist. The `inflections_file` key warning is similar. Wondering if we should keep that one? The other one is a bit useful to because it reminds the user to remove the file if it still exists, although perhaps as part of 2.0 we can just remove all three warnings and include the removal of those keys and files and migrations to `inflections.rb` as part of the upgrade path. Let me know what makes sense to you.